libfwupd/fwupd-enums.h

```diff
@@ -1427,6 +1427,14 @@ typedef enum {
 	 * Since: 1.3.6
 	 */
 	FWUPD_VERSION_FORMAT_DELL_BIOS,
+	/**
+	 * FWUPD_VERSION_FORMAT_DELL_BIOS_MODERN:
+	 *
+	 * Dell BIOS AA.BB.CC style.
```

Comment:
Isn't it AA.BB.CC.XX?

Reply:
unfortunately it's not a quad, it's another interesting format.

Reply:
Old dell bios was `IGNORE.BB.CC.DD` and new dell bios seems to be `AA.BB.CC.IGNORE`. I'm slightly worried that we might try and compare `1.2.3.4` (visible as *1.2.3*) with `1.2.3.5` (visible as *1.2.3*) and get *False* but maybe that's the same problem as comparing dell-bios `1.2.3.4` (visible as *2.3.4*) with `999.2.3.4` (visible as *2.3.4*) although probably less likely.

@CragW I'm okay with a new verfmt (although we need a 1_9_X backport, a LVFS patch, and you also need to make any firmware using the new verfmt depend on a very new fwupd version) -- but I am a bit confused on the bigger-picture *why* --i.e. is this something that the IBV is telling you is going to happen?

I also disagree with @superm1 a bit on the name, so far I've tried to keep the AABB out of the verfmts. How about `dell-msb`, `dell-ignore-lsb`, or `dell-no-lsb` instead?

Reply:
> Old dell bios was `IGNORE.BB.CC.DD` and new dell bios seems to be `AA.BB.CC.IGNORE`. 

LSB for the new format will be 00.

> I'm okay with a new verfmt (although we need a 1_9_X backport, a LVFS patch, and you also need to make any firmware using the new verfmt depend on a very new fwupd version) 

Yup.. that isn't easy to land a new fwupd version on some distro....

> but I am a bit confused on the bigger-picture _why_ --i.e. is this something that the IBV is telling you is going to happen?

The fw_version in ESRT in recent BIOS releases already started using the new verfmt (surprise!), for instance 16777216 is representing 1.0.0 instead previously used 65536. So several dozens already upload to lvfs, got to figure out how to fix verfmt for all of them.

> I also disagree with @superm1 a bit on the name, so far I've tried to keep the AABB out of the verfmts. How about `dell-msb`, `dell-ignore-lsb`, or `dell-no-lsb` instead?

How about dell-bios-msb?

Reply:
> LSB for the new format will be 00

Okay, that makes my _vercmp on the raw value_ concern go away.

> that isn't easy to land a new fwupd version on some distro

Actually, it's not as bad as you think. LVFS has the idea of a "fallback verfmt" so we'd set `dell-bios-msb` with a fallback of `quad` -- then on new fwupd's you'd see the "matching AA.BB.CC" and on older fwupd's you see the AA.BB.CC.DD -- so visually you'd just see an extra `.0` on the UX -- the version comparison logic would still work correctly treating it as a quad afterall.

> The fw_version in ESRT in recent BIOS releases already started using the new verfmt

Sounds like one of your BIOS engineers didn't read the Dell ESRT specification correctly :)

> How about dell-bios-msb

Yes, works for me. Thanks.